File: dailyques/Q865.cpp
//Q865.cpp
class Solution {
public:
    pair<TreeNode*,int> dfs(TreeNode* root){
        if(!root) return {nullptr,0};
        auto L=dfs(root->left), R=dfs(root->right);
        if(L.second==R.second) return {root,L.second+1};
        return (L.second>R.second)? make_pair(L.first,L.second+1)
                                  : make_pair(R.first,R.second+1);
    }
    TreeNode* subtreeWithAllDeepest(TreeNode* root){
        return dfs(root).first;
    }
};

File: dailyques/Q865.py
#Q865.py
class Solution(object):
    def subtreeWithAllDeepest(self, root):
        def dfs(node):
            if not node: return (None,0)
            l,dl=dfs(node.left); r,dr=dfs(node.right)
            if dl==dr: return (node,dl+1)
            return (l,dl+1) if dl>dr else (r,dr+1)
        return dfs(root)[0]
